### ISSUE TITLE
Odd entity selector argument permissions bug

### ISSUE THREAD

**Issue by SHsuperCM** [JorelAli/CommandAPI#176]
**CommandAPI version**
5.4

**Minecraft version**
1.16.3

**Describe the bug**
The entity selector argument seems to work with only explicit player names when sender is not opped.
Other @ selectors are returning 0 entities/players when the command executes.

**My code**
This code reproduces the behavior:
*Tested on multiple commands with entity selector arguments, for all @ selectors
```java
new CommandAPICommand("twisteditems")
                        .withAliases("ti")
                        .withPermission(CommandPermission.fromString("command.twisteditems"))
                        .withArguments(
                                new LiteralArgument("give"),
                                new EntitySelectorArgument("target", EntitySelectorArgument.EntitySelector.MANY_PLAYERS),
                                newTwistedItemArgument())
                        .executes((sender, args) -> {
                            ItemStack itemStack = ((TwistedItem) args[1]).create();
                            for (Player player : ((Collection<Player>) args[0]))
                                player.getInventory().addItem(itemStack);
                            int counter = ((Collection<Player>) args[0]).size();
                            if (counter == 0)
                                sender.sendMessage("No players targeted");
                            else
                                sender.sendMessage(new TextComponent("Gave 1 ["), itemStack.getItemMeta().getDisplayNameComponent()[0], new TextComponent(ChatColor.RESET.toString() + "] to " + (counter == 1 ? ((Player) ((Collection<Player>) args[0]).toArray()[0]).getDisplayName() : counter + " players")));
                        })
                        .register();
```

**Expected behavior**
When not opped(but with proper permissions for the command).
`/ti give <my username> <item>`
to have the same behavior as
`/ti give @s <item>`

**Comment by JorelAli** [comment]
Right.
Simply put, this is caused because the CommandAPI uses the built-in vanilla entity selector argument, which automatically checks if the command sender has the `minecraft.command.selector` permission. Luckily, this doesn't look like a difficult fix (hopefully!). I'll see if I can push out a v5.5 with this fix included in it ASAP.

(Devnote: All it needs is extracting the main function from NMS, getting hold of the underlying instance of `EntitySelector`, applying one reflection call to override the `checkPermissions` member upon that instance and then accessing everything as normal)

**Comment by JorelAli** [comment]
Issue has been fixed in commits 1beda1bdad159ace151e083aa33de00f93a280a8 and e47a006d098c8f0bf8b7c32eb9847589c65bcdf1. This will be included in the v5.5 release (ETA 24 hours)

**Comment by JorelAli** [comment]
@SHsuperCM [CommandAPI v5.5 has been released](https://github.com/JorelAli/CommandAPI/releases/tag/v5.5).
